paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/sink/UnawareBucketWriteSink.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.paimon.flink.sink;
+
+import org.apache.paimon.flink.compact.UnawareBucketCompactionTopoBuilder;
+import org.apache.paimon.operation.Lock;
+import org.apache.paimon.table.AppendOnlyFileStoreTable;
+
+import org.apache.flink.api.common.RuntimeExecutionMode;
+import org.apache.flink.configuration.ExecutionOptions;
+import org.apache.flink.streaming.api.datastream.DataStream;
+import org.apache.flink.streaming.api.datastream.DataStreamSink;
+import org.apache.flink.streaming.api.transformations.PartitionTransformation;
+import org.apache.flink.table.data.RowData;
+
+import java.util.Map;
+
+/**
+ * Build topology for unaware bucket table sink.
+ *
+ * <p>Note: in unaware-bucket mode, we don't shuffle by bucket in inserting. We can assign
+ * compaction to the inserting jobs aside.
+ */
+public class UnawareBucketWriteSink extends FileStoreSink {
+
+    private final boolean enableCompaction;
+    private final AppendOnlyFileStoreTable table;
+
+    public UnawareBucketWriteSink(
+            AppendOnlyFileStoreTable table,
+            Lock.Factory lock,
+            Map<String, String> overwritePartitions,
+            LogSinkFunction logSinkFunction) {
+        super(table, lock, overwritePartitions, logSinkFunction);
+        this.table = table;
+        this.enableCompaction = !table.coreOptions().writeOnly();
+    }
+
+    public DataStreamSink<?> build(DataStream<RowData> input, Integer parallelism) {
+        DataStream<RowData> partitioned = partition(input, parallelism);
+
+        return sinkFrom(partitioned);
+    }
```

Comment:
This `build` method looks strange to me. This is not a builder class. Why not just merge all these into `sinkFrom`?

Reply:
Learned from `RowDynamicBucketSink`.

Reply:
changed this from build to sinkFrom